compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java

```diff
@@ -147,7 +152,35 @@ public static void deletePath(Path path) throws IOException {
 
         Files.delete(path);
     }
-        
+
+    /**
+     * Check if the given image is a PNG.
+     *
+     * @param filePath image file path
+     * @return is valid PNG file
+     * @throws IOException error when reading the given file path
+     */
+    public static boolean isValidPng(Path filePath) throws IOException {
```

Comment:
Shall we change this to a generic method usable to validate any file against any provided magic byte. It's fine to have `isValidPng` as well, but ideally we should extract the core logic into a generic reusable method IMO, which accepts filename, hexvalue (magic byte), and offset.

Reply:
@ayomawdb Can you pls check added `isMatchingImageFormat` method?

Reply:
+1